src/edge/edgeManager.ts

```diff
@@ -271,6 +280,39 @@ export class EdgeManager {
         }
     }
 
+    public async checkAndUpdateASAJob(templateFile: string, moduleName: string) {
+        if (this.asaUpdateStatus === ASAUpdateStatus.Idle) {

```

Comment:
I will prefer here it just a command entry point to populate some necessary information and UI. And the main logic should be moved to asa manager.

Reply:
OK